bigbluebutton-web/grails-app/controllers/org/bigbluebutton/web/controllers/ApiController.groovy

```diff
@@ -1834,13 +1834,22 @@ class ApiController {
     }
   }
 
+  private String subLogoutParams(params, logoutURL) {
+    String newURL = logoutURL;
+    newURL = newURL.replace('%%MEETINGID%%', params.meetingID != null ? params.meetingID : "");
+    newURL = newURL.replace('%%USERID%%', params.userID != null ? params.userID : "");
```

Comment:
There's an inconsistency here!
The function introduced in #21028 replaces `%%USERID%%` with the internal `userId`.
However, in this context, `params.userID` is actually the external `userId`.

I think the function `respondWithRedirect` should receive the `logoutURL` as parameter with the replacements already applied. This would be easier because the caller has the internal `userId` at hand.

----
On the other hand, it might make sense for the `logoutURL` to receive the `externalUserId` in some contexts... In my opinion, it's a good idea to add this new option as a placeholder for `logoutURL`.